### PR TITLE
Implement Create today screen

### PR DESCRIPTION
**Pull request by mateo762**
Implemented the Today Screen, it consists of three parts

- First part
The "today" section, it shows all the habits that are due to today, all the habits that have already been passed ( imagine its 13:00, and there were habits at 9:00) have a line through. They are ordered from the earliest that you have to complete to the latest.

- Second part
The "next up" section, it shows what habit is the next one that you will have to complete, it can be on any other day ( imagine its monday and your next habit is on friday), it will also indicate in how many hours you will have to complete the habit.

- Third part
The "images" section, it shows the latest 3 images of the next up habit.

At the moment the habits and the images are hardcoded, so it is something that needs to be changed

Tests have also been implemented, since the time can change and so the habits that appear in the screen I have taken a fixed time to test ( 21/03/2023 12:55 ), so the habits don't change. I have tested several cases, when there are no habits, when there are no habits in the today section, or when there are habits everywhere.

Screenshot 2023-03-21 151906

### CODE REVIEW

File: app/src/androidTest/java/com/github/mateo762/myapplication/HabitsTestData.kt
Comment (j-encinas) [inline app/src/androidTest/java/com/github/mateo762/myapplication/HabitsTestData.kt:1]: Good idea to have some separate test data. It could be very useful for future tests as well

File: app/src/androidTest/java/com/github/mateo762/myapplication/LoginActivityTest.kt
Comment (j-encinas) [inline app/src/androidTest/java/com/github/mateo762/myapplication/LoginActivityTest.kt:1]: Nice job spotting login and register were grouped in different packages and nice work putting them together

File: app/src/androidTest/java/com/github/mateo762/myapplication/TodayFragmentTest.kt
Comment (j-encinas) [inline app/src/androidTest/java/com/github/mateo762/myapplication/TodayFragmentTest.kt:1]: Looks good, perhaps we could add more tests in the future once the live data functionality is running

File: app/src/main/java/com/github/mateo762/myapplication/Habit.kt
Comment (j-encinas) [inline app/src/main/java/com/github/mateo762/myapplication/Habit.kt:1]: Why the change of names?

File: app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt
Comment (j-encinas) [inline app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt:1]: Nice changes!

File: app/src/main/java/com/github/mateo762/myapplication/ui/habits/CreateHabitScreen.kt
Comment (j-encinas) [inline app/src/main/java/com/github/mateo762/myapplication/ui/habits/CreateHabitScreen.kt:1]: Are there any changes here or is it the same as the other file that was deleted?

File: app/src/main/java/com/github/mateo762/myapplication/ui/home/TodayScreen.kt [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+                                style = if (isCompleted) {
+                                    MaterialTheme.typography.h5.copy(
+                                        textDecoration = TextDecoration.LineThrough
+                                    )
+                                } else {
+                                    MaterialTheme.typography.h5
+                                }
+                            )
+                            Text(
+                                text = "0\uD83D\uDD25",
```
Comment (j-encinas) [inline app/src/main/java/com/github/mateo762/myapplication/ui/home/TodayScreen.kt:132]: The idea of the flame emoji was as a marker of an ongoing habit that is being consistently completed, but maybe we can modify that later, once the live data is usable and the values can be updated

Comment (j-encinas) [reply app/src/main/java/com/github/mateo762/myapplication/ui/home/TodayScreen.kt:132]: Otherwise, looks good to me

File: app/src/main/java/com/github/mateo762/myapplication/authentication/LoginActivity.kt
Comment (j-encinas) [inline app/src/main/java/com/github/mateo762/myapplication/authentication/LoginActivity.kt:1]: Nice refactoring here

File: app/src/main/java/com/github/mateo762/myapplication/habits/fragments/CreateHabitFragment.kt
Comment (j-encinas) [inline app/src/main/java/com/github/mateo762/myapplication/habits/fragments/CreateHabitFragment.kt:1]: Good refactor, but why the file change?

**Review by j-encinas — COMMENTED** [review]
Overall, very good stuff, I noted some things that are minor details, but the feature has tests and is functional (although not with the final data but that's another issue for later). Good refactoring of some weird code that was there from last sprint, and also doing some work on other things strictly outside your task too.

If you can resolve the comments and let me know about CreateHabitScreen.kt and the field names in the Habit class, I'll be happy to approve the merge.

File: app/src/main/java/com/github/mateo762/myapplication/Habit.kt
Comment (mateo762) [reply app/src/main/java/com/github/mateo762/myapplication/Habit.kt:1]: I feel like its redundant that the attributes of Habit start with "habit...", its more concise and short to call "habitName" as "name" directly.

File: app/src/main/java/com/github/mateo762/myapplication/habits/fragments/CreateHabitFragment.kt
Comment (mateo762) [reply app/src/main/java/com/github/mateo762/myapplication/habits/fragments/CreateHabitFragment.kt:1]: I feel like its easier to understand the purpose of the class "CreateHabit" than "DevelopHabit", since at the end you are "creating a habit", not really developing one.

File: app/src/main/java/com/github/mateo762/myapplication/ui/habits/CreateHabitScreen.kt
Comment (mateo762) [reply app/src/main/java/com/github/mateo762/myapplication/ui/habits/CreateHabitScreen.kt:1]: That class is the same as "DevelopHabit", I just changed the class name because I thought the name was more concise and made it easier to understand.

File: app/src/main/java/com/github/mateo762/myapplication/Habit.kt
Comment (j-encinas) [reply app/src/main/java/com/github/mateo762/myapplication/Habit.kt:1]: Okay, I see what you mean

File: app/src/main/java/com/github/mateo762/myapplication/habits/fragments/CreateHabitFragment.kt
Comment (j-encinas) [reply app/src/main/java/com/github/mateo762/myapplication/habits/fragments/CreateHabitFragment.kt:1]: Okay, thanks for the explanation

**Review by j-encinas — APPROVED** [review]
